commit 4fc0d0a89efa
Author: Randell Jesup <rjesup@jesup.org>
Date:   Tue Oct 17 16:27:20 2017 -0400

    Bug 1407680 - Set DelayAgnostic and ExtendedFilter options. r=dminor, a=ritu
    
    This was lost due to API changes in the update to upstream v57
    
    --HG--
    extra : source : 123965af3a606397e7b0f5705ae3365ad64a1258
---
 dom/media/webrtc/MediaEngineWebRTC.cpp      |  7 +++++--
 dom/media/webrtc/MediaEngineWebRTC.h        |  8 +++++++-
 dom/media/webrtc/MediaEngineWebRTCAudio.cpp | 10 +++++++++-
 3 files changed, 21 insertions(+), 4 deletions(-)

diff --git dom/media/webrtc/MediaEngineWebRTC.cpp dom/media/webrtc/MediaEngineWebRTC.cpp
index 772b9bb6e52f..e8429506dcae 100644
--- dom/media/webrtc/MediaEngineWebRTC.cpp
+++ dom/media/webrtc/MediaEngineWebRTC.cpp
@@ -111,6 +111,8 @@ MediaEngineWebRTC::MediaEngineWebRTC(MediaEnginePrefs &aPrefs)
     mVoiceEngine(nullptr),
     mAudioInput(nullptr),
     mFullDuplex(aPrefs.mFullDuplex),
+    mDelayAgnostic(aPrefs.mDelayAgnostic),
+    mExtendedFilter(aPrefs.mExtendedFilter),
     mHasTabVideoSource(false)
 {
   nsCOMPtr<nsIComponentRegistrar> compMgr;
@@ -294,7 +296,7 @@ MediaEngineWebRTC::EnumerateAudioDevices(dom::MediaSourceEnum aMediaSource,
 #endif
 
   if (!mVoiceEngine) {
-    mVoiceEngine = webrtc::VoiceEngine::Create(/*mConfig*/);
+    mVoiceEngine = webrtc::VoiceEngine::Create();
     if (!mVoiceEngine) {
       return;
     }
@@ -367,7 +369,8 @@ MediaEngineWebRTC::EnumerateAudioDevices(dom::MediaSourceEnum aMediaSource,
         audioinput = new mozilla::AudioInputCubeb(mVoiceEngine, i);
       }
       aSource = new MediaEngineWebRTCMicrophoneSource(mVoiceEngine, audioinput,
-                                                      i, deviceName, uniqueId);
+                                                      i, deviceName, uniqueId,
+                                                      mDelayAgnostic, mExtendedFilter);
       mAudioSources.Put(uuid, aSource); // Hashtable takes ownership.
       aASources->AppendElement(aSource);
     }
diff --git dom/media/webrtc/MediaEngineWebRTC.h dom/media/webrtc/MediaEngineWebRTC.h
index 6b636adf396d..4567a5c0211a 100644
--- dom/media/webrtc/MediaEngineWebRTC.h
+++ dom/media/webrtc/MediaEngineWebRTC.h
@@ -501,7 +501,9 @@ public:
                                     mozilla::AudioInput* aAudioInput,
                                     int aIndex,
                                     const char* name,
-                                    const char* uuid);
+                                    const char* uuid,
+                                    bool aDelayAgnostic,
+                                    bool aExtendedFilter);
 
   void GetName(nsAString& aName) const override;
   void GetUUID(nsACString& aUUID) const override;
@@ -624,6 +626,8 @@ private:
 
   int mCapIndex;
   int mChannel;
+  bool mDelayAgnostic;
+  bool mExtendedFilter;
   MOZ_INIT_OUTSIDE_CTOR TrackID mTrackID;
   bool mStarted;
 
@@ -678,6 +682,8 @@ private:
   webrtc::VoiceEngine* mVoiceEngine;
   RefPtr<mozilla::AudioInput> mAudioInput;
   bool mFullDuplex;
+  bool mDelayAgnostic;
+  bool mExtendedFilter;
   bool mHasTabVideoSource;
 
   // Store devices we've already seen in a hashtable for quick return.
diff --git dom/media/webrtc/MediaEngineWebRTCAudio.cpp dom/media/webrtc/MediaEngineWebRTCAudio.cpp
index ff77065b09fa..a8bd52f01b24 100644
--- dom/media/webrtc/MediaEngineWebRTCAudio.cpp
+++ dom/media/webrtc/MediaEngineWebRTCAudio.cpp
@@ -191,13 +191,17 @@ MediaEngineWebRTCMicrophoneSource::MediaEngineWebRTCMicrophoneSource(
     mozilla::AudioInput* aAudioInput,
     int aIndex,
     const char* name,
-    const char* uuid)
+    const char* uuid,
+    bool aDelayAgnostic,
+    bool aExtendedFilter)
   : MediaEngineAudioSource(kReleased)
   , mVoiceEngine(aVoiceEnginePtr)
   , mAudioInput(aAudioInput)
   , mMonitor("WebRTCMic.Monitor")
   , mCapIndex(aIndex)
   , mChannel(-1)
+  , mDelayAgnostic(aDelayAgnostic)
+  , mExtendedFilter(aExtendedFilter)
   , mTrackID(TRACK_NONE)
   , mStarted(false)
   , mSampleFrequency(MediaEngine::DEFAULT_SAMPLE_RATE)
@@ -781,6 +785,10 @@ MediaEngineWebRTCMicrophoneSource::InitEngine()
   mVoEBase = webrtc::VoEBase::GetInterface(mVoiceEngine);
 
   mVoEBase->Init();
+  webrtc::Config config;
+  config.Set<webrtc::ExtendedFilter>(new webrtc::ExtendedFilter(mExtendedFilter));
+  config.Set<webrtc::DelayAgnostic>(new webrtc::DelayAgnostic(mDelayAgnostic));
+  mVoEBase->audio_processing()->SetExtraOptions(config);
 
   mVoERender = webrtc::VoEExternalMedia::GetInterface(mVoiceEngine);
   if (mVoERender) {
